### PR TITLE
GUVNOR-2317: Wrong Commit History layout for long commit messages

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2317

Tidied up separation between M, V and P; fixed the CSS issues and wrote some tests.

### CODE REVIEW

File: guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/edit/RepositoryEditorView.java [hunk 1]
```diff
@@ -145,9 +132,7 @@ public void execute( final VersionRecord record ) {
         loadMore.addClickHandler( new ClickHandler() {
             @Override
             public void onClick( ClickEvent event ) {
-                if ( commitNavigator != null ) {
-                    presenter.getLoadMoreHistory( commitNavigator.getLastIndex() );
-                }
+                presenter.getLoadMoreHistory( history.getWidgetCount() );
```
Comment (Rikkola) [inline guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/edit/RepositoryEditorView.java]: I have used the presenter.on*() format for any action that is done to presenter from the view impl. For example a refresh button click would be presenter.onRefresh(), not caring about what the actual widget is, but on what the action is. 
Just a thing I do to try and keep the readability better for the presenter methods and I feel it also helps reading the unit tests. 
Just a proposal for what I think we should do. Now that I think these methods should probably also be package private.

File: guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/navigator/CommitNavigatorEntry.java [hunk 2]
```diff
@@ -36,87 +37,46 @@
...
+    }
 
-        navigator.setWidget( row, col, new Widget() {{
-            setElement( messageCol );
-        }} );
+    private void initWidget( final VersionRecord versionRecord,
+                             final ParameterizedCommand<VersionRecord> onRevertCommand ) {
+        message.getElement().setInnerSafeHtml( makeSafeHtml( versionRecord.comment() ) );
+        author.getElement().setInnerSafeHtml( makeSafeHtml( versionRecord.author() + " - " ) );
+        date.getElement().setInnerSafeHtml( makeSafeHtml( fmt.format( versionRecord.date() ) ) );
```
Comment (Rikkola) [inline guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/navigator/CommitNavigatorEntry.java]: Way better to use UiBinder here, but would it be possible to use a HTML element or just a Label in the ui.xml for these? 
There might be a reason for doing it this way and I am missing it. Just feels more complex and if I just look at the ui.xml it is not clear what is shown there.

**Comment by Rikkola** [comment]
+1

**Comment by csadilek** [comment]
Thanks.

master: 664e3743df7995e2abfcb8ad54a49b20eace04d3

**Pull request closed without merging**